pkg/operator/ceph/cluster/predicate.go

```diff
@@ -132,23 +132,19 @@ func watchControllerPredicate(rookContext *clusterd.Context) predicate.Funcs {
 				}
 				diff := cmp.Diff(objOld.Spec, objNew.Spec, resourceQtyComparer)
 				if diff != "" {
-					logger.Infof("CR has changed for %q. diff=%s", objNew.Name, diff)
-
-					// If spec Images are different we cancel any on-going orchestration
-					if objOld.Spec.CephVersion.Image != objNew.Spec.CephVersion.Image {
-						// Set the cancellation flag to stop any ongoing orchestration
-						rookContext.RequestCancelOrchestration.Set()
-
-						logger.Info("upgrade requested, cancelling any ongoing orchestration")
-					}
+					// Set the cancellation flag to stop any ongoing orchestration
+					rookContext.RequestCancelOrchestration.Set()
```

Comment:
This is related to the new design. Instead of having to keep a timeout on the OSDs, if we cancel on every CR change, then we don't have to worry about long OSD reconciles starving other updates.

Reply:
I still don't understand why this is necessary, maybe I've missed something in the design but I don't recall this part.

Reply:
We have 2 options to make sure that OSD reconciliation does not stop Rook from reconciling other resources when there are CephCluster CR updates:
1) Cancel orchestration every time the CephCluster CR is updated so that any ongoing OSD reconcile will be soon stopped. The OSD reconcile will resume on the next CR reconcile. (the option implemented here)
2) Put a timeout on the OSD reconcile that will stop it when progress hasn't been made for a while. Return an error so the OSDs can be reconciled again if there are no CR updates. If there are CR updates, now is when they will take hold.

In huddle conversations, Travis and I came to the agreement that stopping the reconcile on any CR update (not just Ceph version updates/upgrades) is preferred. User changes will take place more quickly, and we don't have to complicate the OSD design/code with a timeout.

Reply:
I see, it makes sense. I guess I was worried that could trigger undesired behavior if we brutally abort every and any reconcile. However, if it occurs it's probably a bug we should fix. Thanks!

Reply:
Agreed. As a note, I haven't seen any reconcile-storm or similar events that result in restarting the reconcile accidentally in my tests.